llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2393,7 +2393,7 @@ template <> Value *SPIRVToLLVM::transValueWithOpcode<OpLoad>(SPIRVValue *const s
     }
   }
 
-  bool isVolatile = spvLoad->SPIRVMemoryAccess::isVolatile(true);
+  bool isVolatile = static_cast<bool>(spvLoad->SPIRVMemoryAccess::isVolatile(true)) | spvLoad->getSrc()->isVolatile();
```

Comment:
As discussed with Nicolai, we conditionally add the check of spvLoad->getSrc()->isVolatile() for robustness2 to not prevent beneficial optimizations of LLVM.

Reply:
Fixed and updated the change based on #746.